EtherealCereal/EtherealCereal.swift

```diff
@@ -118,14 +118,23 @@ public extension Data {
 }
 
 public extension String {
+
     public var hexadecimalData: Data? {
-        let str: String
-        let offset = self.index(self.startIndex, offsetBy: 2)
-        if self.substring(to: offset) == "0x" {
-            str = self.substring(from: offset)
-        } else {
-            str = self
+
+        let regex: NSRegularExpression
+        do {
+            regex = try NSRegularExpression(pattern: "^(?:0x)?([a-fA-F0-9]*)$", options: .caseInsensitive)
+        } catch {
+            fatalError("Invalid regular expression pattern")
```

Comment:
Do you want this to stop the program entirely if the regex doesn't compile, even in production? If not, consider using `assertionFailure`, which no-ops in production. If you'd rather it fail SUPER loudly, then this works.

Reply:
the pattern is static, and if it doesn't compile then a lot of core functionality will not work, so i think failing loudly is a good idea.

But i'd also prefer it not need to be compiled with every call. in other languages i'd compile the regex at boot time, and reuse it each time. would this make sense in swift as well?

e.g. something like this declared outside of the class:
`let HEX_DATA_RE = try ! NSRegularExpression(pattern: "^(?:0x)?([a-fA-F0-9]*)$", options: .caseInsensitive)`
(not sure about the syntax here, or the `try!`)